packages/node-provider/test/unit/node-provider.spec.ts

```diff
@@ -1,7 +1,54 @@
 import NodeProvider from "../../src/node-provider";
-import { NodeMessage, NodeMessageType } from "../../src/types";
+import { NodeMessageType } from "../../src/types";
+
+const originalAddEventListener = window.addEventListener;
+const originalPostMessage = window.postMessage;
 
 describe("NodeProvider", () => {
+  beforeAll(() => {
+    let messageCallback;
+    window.addEventListener = jest.fn((eventName, callback) => {
+      if (eventName !== "message") {
+        originalAddEventListener(eventName, callback);
+        return;
+      }
+
+      messageCallback = callback;
+    });
+    window.postMessage = jest.fn((message, target, transferables) => {
+      messageCallback({
+        data: message,
+        ports: transferables,
+        type: "message",
+        lastEventId: "0",
+        origin: "localhost",
+        source: null,
+        bubbles: false,
+        cancelBubble: false,
+        cancelable: false,
+        composed: false,
+        currentTarget: window,
+        defaultPrevented: false,
+        eventPhase: 4,
+        isTrusted: true,
+        returnValue: null,
+        srcElement: window.document.body,
+        target: window.document.body,
+        timeStamp: Date.now(),
+        composedPath() {
+          return [] as EventTarget[];
+        },
+        initEvent() {},
+        preventDefault() {},
+        stopImmediatePropagation() {},
+        stopPropagation() {},
+        AT_TARGET: 0,
+        BUBBLING_PHASE: 1,
+        CAPTURING_PHASE: 2,
+        NONE: 3
+      });
```

Comment:
What's up with this huge object? Maybe needs some documentation

Reply:
It's a mock for a MessageEvent instance, it's basically everything an event has when it's triggered via `addEventListener`.